### PR TITLE
Limit Surface bounding error

### LINKED ISSUE #1216: [TASK] Limit Surface Integral Bounding Error
--------
Issue Description
--------

**Is your feature request related to a problem? Please describe.**

The Limit Surface integral should allow (optionally) to request for the estimation of the bounding error (maximum error in Probability)

**Describe the solution you'd like**
N/A

**Describe alternatives you've considered**
N/A

----------------
For Change Control Board: Issue Review
----------------
This review should occur before any development is performed as a response to this issue.
- [x] 1. Is it tagged with a type: defect or task?
- [x] 2. Is it tagged with a priority: critical, normal or minor?
- [x] 3. If it will impact requirements or requirements tests, is it tagged with requirements?
- [x] 4. If it is a defect, can it cause wrong results for users? If so an email needs to be sent to the users.
- [x] 5. Is a rationale provided? (Such as explaining why the improvement is needed or why current code is wrong.)

-------
For Change Control Board: Issue Closure
-------
This review should occur when the issue is imminently going to be closed.
- [x] 1. If the issue is a defect, is the defect fixed?
- [x] 2. If the issue is a defect, is the defect tested for in the regression test system?  (If not explain why not.)
- [x] 3. If the issue can impact users, has an email to the users group been written (the email should specify if the defect impacts stable or master)?
- [x] 4. If the issue is a defect, does it impact the latest release branch? If yes, is there any issue tagged with release (create if needed)?
- [x] 5. If the issue is being closed without a pull request, has an explanation of why it is being closed been provided?

### PR DESCRIPTION
**Pull request by alfoa**
--------
Pull Request Description
--------
##### What issue does this change request address? (Use "#" before the issue to link it, i.e., #42.)

Closes #1216

##### What are the significant changes in functionality due to this change request?

This PR adds the optional possibility to estimate the bounding error of the limit surface integral.

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [x] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added?
- [x] 9. If any test used as a basis for documentation examples (currently found in `raven/tests/framework/user_guide` and `raven/docs/workshop`) have been changed, the associated documentation must be reviewed and assured the text matches the example.

### CODE REVIEW

File: doc/user_manual/postprocessor.tex [hunk 1]
```diff
@@ -809,6 +809,16 @@ \subsubsection{LimitSurfaceIntegral}
      \item  \xmlNode{integralType}, \xmlDesc{string, optional field}, specifies the type of integrations that
                 need to be used. Currently only MonteCarlo integration is available
                 \default{MonteCarlo}
+    \item  \xmlNode{computeBounds}, \xmlDesc{bool, optional field}, 
+    activates the computation of the bounding error of the limit 
+    surface integral ( maximum error in the identification of the 
+    limit surface location). If True, the bounding error is stored 
+    in a variable named as \xmlNode{outputName} appending the suffix
+    ``_err''. For example, if \xmlNode{outputName} is
```
Comment (wangcj05) [inline doc/user_manual/postprocessor.tex:817]: ``_err'' -> ```\_err```

File: doc/user_manual/postprocessor.tex [hunk 2]
```diff
@@ -809,6 +809,16 @@ \subsubsection{LimitSurfaceIntegral}
...
                 need to be used. Currently only MonteCarlo integration is available
                 \default{MonteCarlo}
+    \item  \xmlNode{computeBounds}, \xmlDesc{bool, optional field}, 
+    activates the computation of the bounding error of the limit 
+    surface integral ( maximum error in the identification of the 
+    limit surface location). If True, the bounding error is stored 
+    in a variable named as \xmlNode{outputName} appending the suffix
+    ``_err''. For example, if \xmlNode{outputName} is 
+    ``EventProbability'', the bounding error will be stored as 
+    ``EventProbability_err'' (this variable name must be listed as
```
Comment (wangcj05) [inline doc/user_manual/postprocessor.tex:819]: ``EventProbability_err'' -> ```EventProbability\_err```

File: framework/PostProcessors/LimitSurfaceIntegral.py [hunk 3]
```diff
@@ -93,8 +97,10 @@ def __init__(self, messageHandler):
     self.integralType = 'montecarlo'  # integral type (which alg needs to be used). Either montecarlo or quadrature(quadrature not yet)
     self.seed = 20021986  # seed for montecarlo
     self.matrixDict = {}  # dictionary of arrays and target
+    self.computeErrrorBounds = False #  compute the bounding error?
     self.lowerUpperDict = {}
     self.functionS = None
+    self.errorModel = None
```
Comment (wangcj05) [inline framework/PostProcessors/LimitSurfaceIntegral.py:103]: please provide comments for each self variable

File: tests/framework/PostProcessors/LimitSurface/test_LimitSurface_with_err_bounds.xml [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+    <created>2020-04-15</created>
+    <classesTested>Models.PostProcessors.LimitSurface, Models.PostProcessors.LimitSurfaceIntegral</classesTested>
+    <description>
+       This test is aimed to check the capability of RAVEN to compute the integral of the Limit Surface (e.g.
+       Failure probability) reporting the bounding error as well (the maximum error, in probability, in the computation of the limit surface )
+    </description>
+    <revisions>
+      <revision author="alfoa" date="2020-04-15">Added test for computation of the error bound</revision>
+    </revisions>
+    <requirements>R-RA-7</requirements>
```
Comment (wangcj05) [inline tests/framework/PostProcessors/LimitSurface/test_LimitSurface_with_err_bounds.xml:15]: Do you need to update the requirements_list.xml?

Comment (alfoa) [reply tests/framework/PostProcessors/LimitSurface/test_LimitSurface_with_err_bounds.xml]: you are right. I added a new requirement.

**Comment by alfoa** [comment]
This PR adds a new requirement.
As chair of the COB, I approve such addition.

File: framework/PostProcessors/LimitSurfaceIntegral.py (hunk 3 above)
Comment (alfoa) [reply framework/PostProcessors/LimitSurfaceIntegral.py]: done

File: doc/user_manual/postprocessor.tex (hunk 1 above)
Comment (alfoa) [reply doc/user_manual/postprocessor.tex]: fixed

File: doc/user_manual/postprocessor.tex (hunk 2 above)
Comment (alfoa) [reply doc/user_manual/postprocessor.tex]: fixed

**Comment by alfoa** [comment]
@wangcj05 I had to modify the way to compute the error bound (Actually just the perturbation on one of the sides of the LS...). The previous approach was not stable since it was not considering the magnitude of the FOM of interest.

**Comment by wangcj05** [comment]
Checklist is satisfied, and this PR can be merged.

**Comment by wangcj05** [comment]
@alfoa Do you want to delete your branch?

**Pull request merged**